presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/FloatingPointColumnValidator.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    @Override
+    public List<SingleColumn> generateChecksumColumns(Column column)
+    {
+        Expression doubleColumn = column.getCategory().equals(DOUBLE) ? identifier(column.getName()) : new Cast(identifier(column.getName()), DOUBLE.getDisplayName());
+        Expression positiveInfinity = new FunctionCall(QualifiedName.of("infinity"), ImmutableList.of());
+        Expression negativeInfinity = new ArithmeticUnaryExpression(MINUS, positiveInfinity);
+
+        return ImmutableList.of(
+                new SingleColumn(
+                        new FunctionCall(
+                                QualifiedName.of("sum"),
+                                Optional.empty(),
+                                Optional.of(new FunctionCall(QualifiedName.of("is_finite"), ImmutableList.of(identifier(column.getName())))),
+                                Optional.empty(),
+                                false,
+                                ImmutableList.of(doubleColumn)),
+                        Optional.of(identifier(getSumColumnAlias(column)))),
+                new SingleColumn(
+                        new FunctionCall(
+                                QualifiedName.of("count"),
+                                Optional.empty(),
+                                Optional.of(new FunctionCall(QualifiedName.of("is_nan"), ImmutableList.of(identifier(column.getName())))),
+                                Optional.empty(),
+                                false,
+                                ImmutableList.of(identifier(column.getName()))),
+                        Optional.of(identifier(getNanCountColumnAlias(column)))),
+                new SingleColumn(
+                        new FunctionCall(
+                                QualifiedName.of("count"),
+                                Optional.empty(),
+                                Optional.of(new ComparisonExpression(EQUAL, identifier(column.getName()), positiveInfinity)),
+                                Optional.empty(),
+                                false,
+                                ImmutableList.of(identifier(column.getName()))),
+                        Optional.of(identifier(getPositiveInfinityCountColumnAlias(column)))),
+                new SingleColumn(
+                        new FunctionCall(
+                                QualifiedName.of("count"),
+                                Optional.empty(),
+                                Optional.of(new ComparisonExpression(EQUAL, identifier(column.getName()), negativeInfinity)),
+                                Optional.empty(),
+                                false,
+                                ImmutableList.of(identifier(column.getName()))),
+                        Optional.of(identifier(getNegativeInfinityCountColumnAlias(column)))));
+    }
+
+    @Override
+    boolean isMatched(Column column, ChecksumResult controlResult, ChecksumResult testResult)
```

Comment:
Do you feel it will be useful to return the result of validation in a better way? Roughly, the following is what I am thinking of:

```
class ColumnMatchResult {
  boolean isMatch;
  Optional<String> errorMessage;
}
```

The idea being to be able to return custom exception messages. Example - `The sums of test (1234.43545) and control (1345.34353) were off by x%` So that the criteria for test failure is clear.

Reply:
This will be useful, but I would delay it to next iteration, as it make the output more complicated. We need to think about how we're going to represent them in the output event, as well as, how we're going to store and present the data to the UI

Reply:
We already have a `errorMessage` field - so we can just put it as text there. We can have that in the first iteration.

Reply:
The `errorMessage` field is parsed by some php code to present for the UI at this time, so we probably should not break its contract here.